scripts/inc.updatebinary.sh

```diff
@@ -982,15 +986,21 @@ for gapp_name in $core_gapps_list; do
   get_appsize "Core/$gapp_name";
   core_size=$((core_size + appsize));
 done;
-unzip -o "$ZIP" "Optional/keybd_lib.tar.xz" -d /tmp;
-keybd_lib_size=$(tar -tvJf "/tmp/Optional/keybd_lib.tar.xz" "keybd_lib" 2>/dev/null | awk 'BEGIN { app_size=0; } { file_size=$3; app_size=app_size+file_size; } END { printf "%.0f\n", app_size / 1024; }');
-rm -f "/tmp/Optional/keybd_lib.tar.xz";
 
-# Determine final size of Core Apps
+# Add Keyboard Lib size to core, if it exists
 if ( ! contains "$gapps_list" "keyboardgoogle" ); then
+  unzip -o "$ZIP" "Optional/keybd_lib.tar.xz" -d /tmp;
+  keybd_lib_size=$(tar -tvJf "/tmp/Optional/keybd_lib.tar.xz" "keybd_lib" 2>/dev/null | awk 'BEGIN { app_size=0; } { file_size=$3; app_size=app_size+file_size; } END { printf "%.0f\n", app_size / 1024; }');
+  rm -f "/tmp/Optional/keybd_lib.tar.xz";
   core_size=$((core_size + keybd_lib_size)); # Add Keyboard Lib size to core, if it exists
 fi
 
+# Do not touch AOSP keyboard if it's niether removed or replaced by Google's one
+if ( ! contains "$gapps_list" "keyboardgoogle" ) && ( ! contains "$gapps_removal_list" "keyboardstock" ) && [ "$keybd_lib_upgrade" != "true" ]; then
```

Comment:
Are we sure this also goes well then for users "upgrading" their GApps package to the most recent version? They did receive the lib previously from our install, and it should not accidentally be removed by the gapps-remove list.